test/guard_intf_test.cpp

```diff
@@ -55,59 +63,38 @@ TEST_F(TestGuardRecord, CreateGuardRecord)
 {
     //! TODO need to test serial number and part number once device tree support
     //! is available.
-    openpower::guard::libguard_init();
-    std::string phyPath = "/sys-0/node-0/proc-1/eq-0/fc-0/core-0";
-    std::optional<openpower::guard::EntityPath> entityPath =
-        openpower::guard::getEntityPath(phyPath);
-    EXPECT_NE(entityPath, std::nullopt);
-    openpower::guard::create(*entityPath);
-    openpower::guard::GuardRecords records = openpower::guard::getAll();
+    auto [records, entityPath] =
+        LibGuardInit("/sys-0/node-0/proc-1/eq-0/fc-0/core-0");
     EXPECT_EQ(records.size(), 1);
     openpower::guard::GuardRecord record = records.at(0);
     EXPECT_EQ(record.targetId, entityPath);
 }
 
 TEST_F(TestGuardRecord, ClearGuardGoodPathTest)
 {
-    openpower::guard::libguard_init();
-    std::string phyPath = "/sys-0/node-0/proc-1/eq-0/fc-0/core-0";
-    std::optional<openpower::guard::EntityPath> entityPath =
-        openpower::guard::getEntityPath(phyPath);
-    EXPECT_NE(entityPath, std::nullopt);
-    openpower::guard::create(*entityPath);
+    LibGuardInit("/sys-0/node-0/proc-1/eq-0/fc-0/core-0");
     openpower::guard::clearAll();
     openpower::guard::GuardRecords records = openpower::guard::getAll();
     EXPECT_EQ(records.size(), 0);
 }
 
 TEST_F(TestGuardRecord, DeleteGuardGoodPathTest)
 {
-    openpower::guard::libguard_init();
-    std::string phyPath = "/sys-0/node-0/proc-1/eq-0/fc-0/core-0";
-    std::optional<openpower::guard::EntityPath> entityPath =
-        openpower::guard::getEntityPath(phyPath);
-    openpower::guard::create(*entityPath);
-    phyPath = "/sys-0/node-0/proc-1/eq-0/fc-0/core-1";
-    entityPath = openpower::guard::getEntityPath(phyPath);
-    openpower::guard::create(*entityPath);
-    phyPath = "/sys-0/node-0/proc-0/eq-0/fc-0/core-0";
-    entityPath = openpower::guard::getEntityPath(phyPath);
-    openpower::guard::create(*entityPath);
-    phyPath = "/sys-0/node-0/proc-0/eq-0/fc-0/core-1";
-    entityPath = openpower::guard::getEntityPath(phyPath);
-    openpower::guard::create(*entityPath);
+    LibGuardInit("/sys-0/node-0/proc-1/eq-0/fc-0/core-0");
```

Comment:
Here the libguard Init gets called 4 times whereas it was only called once earlier. Better to avoid it